flytekit/core/base_task.py

```diff
@@ -524,7 +524,7 @@ def dispatch_execute(
                 try:
                     literals[k] = TypeEngine.to_literal(exec_ctx, v, py_type, literal_type)
                 except Exception as e:
-                    raise AssertionError(f"failed to convert return value for var {k}") from e
+                    raise AssertionError(f"failed to convert return value for var {k} with error: {e}") from e
```

Comment:
@wild-endeavor I kept `from e` because I think it'll useful when locally executing tasks